### PR TITLE
Post cache graphql fixes

### PR DESCRIPTION
**Pull request by miratepuffin**
### What changes were proposed in this pull request?
- [x] I think we're using the wrong kind of Result in this file, probably `Result<_, GraphError>` is more appropriate
    - [x] _Originally posted by @fabianmurariu in https://github.com/Pometry/Raphtory/pull/1656#discussion_r1689718817_
- [x] GqlGraph doesn't need a name it can be created from the Path
- [x] `save_graphs_to_work_dir` can be moved into the test module
- [x] `get_graph_name` needs to user .display.to_string to remove the Result and swap to file_stem to remove the file extension
- [x] Any references to `work_dir` inside of `Data` can just used the stored value 
- [x] On currently dead `load_graph_from_path` add the following comments
    - [x] Can we change this to an Option which would also allow renaming of the graph on copy - for namespace
    - [x] We can remove is_disk_graph_dir as we check if the path is a valid diskgraph within load_disk_graph 
- [x] On currently dead `load_disk_graph_from_path` need to change the output to just Result<PathBuf>
- [x] update `get_graph` as per @fabianmurariu comment 
    - [x] I think you want to call get_with(&name, || get_graph_from_path..) here this will make sure you only load the graph once when called by multiple threads
- [x] Is there a nicer way to check for Path correcteness within `contract_graph_full_path`

as per #1705

### Why are the changes needed?
Required as part of the continued cleanup of the graphql server

### Does this PR introduce any user-facing change? If yes is this documented?
No, were changed previously 

### How was this patch tested?
Via current and new unit tests

### Are there any further changes required?
Yes see issue

### CODE REVIEW

File: python/tests/test_graphql.py [hunk 1]
```diff
@@ -216,13 +216,13 @@ def assert_graph_fetch(path):
         assert os.path.exists(expected_path)
         assert_graph_fetch(path)
 
-        path = "./shivam/investigation/g"
+        path = "shivam/investigation/g"
```
Comment (ljeub-pometry) [inline python/tests/test_graphql.py:219]: `"./something"` should fail now right? Can you add a test for that?

File: raphtory-graphql/src/data.rs [hunk 2]
```diff
@@ -285,40 +269,43 @@ fn is_disk_graph_dir(path: &Path) -> bool {
...
-    path.file_name()
-        .and_then(|os_str| os_str.to_str())
-        .map(|str_slice| str_slice.to_string())
-        .ok_or("No file name found in the path or invalid UTF-8")
+pub(crate) fn get_graph_name(path: &Path) -> String {
+    match path.file_stem() {
+        None => path.display().to_string(), //This will only happen if there is no file name
+        Some(stem) => {
+            match stem.to_str().map(|str| str.to_string()) {
+                None => path.display().to_string(), //This should never happen
```
Comment (ljeub-pometry) [inline raphtory-graphql/src/data.rs:277]: this should probably be an error?

File: raphtory-graphql/src/data.rs [hunk 3]
```diff
@@ -285,40 +269,43 @@ fn is_disk_graph_dir(path: &Path) -> bool {
...
 }
 
-pub(crate) fn get_graph_name(path: &Path) -> Result<String, &'static str> {
-    path.file_name()
-        .and_then(|os_str| os_str.to_str())
-        .map(|str_slice| str_slice.to_string())
-        .ok_or("No file name found in the path or invalid UTF-8")
+pub(crate) fn get_graph_name(path: &Path) -> String {
+    match path.file_stem() {
+        None => path.display().to_string(), //This will only happen if there is no file name
```
Comment (ljeub-pometry) [inline raphtory-graphql/src/data.rs:274]: What happens for disk graph currently? In that case the path is a directory which may fall in here?

Comment (ljeub-pometry) [reply raphtory-graphql/src/data.rs:274]: Or are they being handled via bincode now?

File: raphtory-graphql/src/data.rs [hunk 4]
```diff
@@ -753,3 +582,233 @@ pub(crate) mod data_tests {
         );
     }
 }
+
```
Comment (ljeub-pometry) [inline raphtory-graphql/src/data.rs:585]: This is a lot of commented out code, should we maybe park this in a branch instead?

File: raphtory-graphql/src/model/graph/graph.rs [hunk 5]
```diff
@@ -368,7 +314,7 @@ impl GqlGraph {
     ////////////////////////
 
     async fn name(&self) -> String {
-        self.name.clone()
+        get_graph_name(&self.path)
```
Comment (ljeub-pometry) [inline raphtory-graphql/src/model/graph/graph.rs:317]: add a test for nested paths with disk graph for this

**Review by ljeub-pometry — CHANGES_REQUESTED** [review]
I think `name` on DiskGraph is currently returning the full path instead of only the last directory, add a test and fix

File: raphtory-graphql/src/model/graph/graph.rs (hunk 5 above)
Comment (miratepuffin) [reply raphtory-graphql/src/model/graph/graph.rs:317]: We need to add many things for DiskGraphs - as upload etc not currently working - we can do as a followup once the cachegraph is in

**Review by ljeub-pometry — APPROVED** [review]
LGTM

**Pull request merged**